frontend/src/parser/crCalculator.ts

```diff
@@ -0,0 +1,110 @@
+import type { FeatureEntity, Saves } from "../../../shared"
+import { parseDescIntoAutomation } from "./utils"
+
+export function scrapeFeatures(features: FeatureEntity[] = []): number[]{
+    let bonuses: number[] = []
+    let damages: number[] = []
+    let saveDC: number[] = []
+
+    for (const feature of features){
+        let automation: object | null
+
+        if (feature.automation){
+            automation = feature.automation
+        } else {
+            automation = parseDescIntoAutomation(feature.description, feature.name ,1)[0]
+        }
+
+        const bonus = automation ?  findAttackBonus(automation) : undefined
+        const damage = automation ? findAttackDamage(automation) : undefined
+        const dc = automation ? findDC(automation): undefined
+
+        if (bonus && !isNaN(bonus)){
+            bonuses.push(bonus)
+        }
+
+        if (damage && !isNaN(damage)){
+            damages.push(damage)
+        }
+
+        if (dc && !isNaN(dc)){
+            saveDC.push(dc)
+        }
+    }
+
+    const averageAttackBonus = averageValue(bonuses)
+    const averageDPR = averageValue(bonuses)
+    const averageDC = averageValue(saveDC)
+
+    return [averageAttackBonus, averageDPR, averageDC]
+}
+
+export function countProficientSaves(saves: Saves): number {
+    let totalCount = 0
+
+    for (const [key, save] of Object.entries(saves)){
+        if (save.isProficient) totalCount ++
+    }
+
+    return totalCount
+}
+
+export function averageValue(values: number[] = []): number{
+    return Math.round(values
+        .filter(value => !isNaN(value))
+        .reduce((runningTotal, currentValue) => runningTotal + currentValue,0) / values.filter(value => !isNaN(value)).length)
+}
+
+function findAttackBonus(obj: any): number | undefined{
+    if (typeof obj === "object" && obj !== null){
```

Comment:
See my previous comment about using a guard clause instead of an if statement everything is nested in. The same goes for the if attackbonus section! 
```js
if ('attackbonus' in obj) return ...
for (const key ....
```